forest/bonsai/simulate_gps_data.py

```diff
@@ -556,7 +559,21 @@ def calculate_trip(self, origin: Tuple[float, float],
         if coords_str in self.trips.keys():
             path = self.trips[coords_str]
         else:
-            path, _ = get_path(origin, destination, transport, api_key)
+            for try_no in range(3):
+                try:
```

Comment:
It's cleaner to use a `try ... except ... else ...` here. Your `else` block would have:
```
path = get_basic_path(path, transport)
self.trips[coords_str] = path
break
```